tritium-core/src/main/java/com/palantir/tritium/event/AbstractInvocationEventHandler.java

```diff
@@ -68,6 +74,28 @@ public final boolean isEnabled() {
         return isEnabledSupplier.getAsBoolean();
     }
 
+    /**
+     * Returns true if the specified invocation context is not null, otherwise false.
+     *
+     * @param context invocation context
+     * @return true if the specified context is not null, otherwise false
+     */
+    @CheckReturnValue
+    protected final boolean isNonNullContext(@Nullable InvocationContext context) {
+        if (context != null) {
+            return true;
+        }
+
+        logger.debug(
+                "{} encountered null metric context likely due to exception in preInvocation",
+                safeClassName(getClass()));
+        return false;
+    }
+
+    private static SafeArg<String> safeClassName(Object obj) {
+        return SafeArg.of("class", (obj == null) ? "" : obj.getClass().getName());
```

Comment:
Should we log the Class object directly here?

I don't think `obj` is nullable since we use the result of getClass.

Thoughts on making this function responsible for the log.debug, and potentially adding a enabled check around it? That way we'll both keep isNonNullContext below 35b and avoid SafeArg creation for a debug line unless debug logging is enabled.

Reply:
We should be able to use the `@NotNull` and `@Nullable` annotations and have them strictly enforced by Uber's nullaway stuff https://github.com/palantir/tritium/blob/develop/build.gradle#L112-L121

Reply:
@iamdanfox I had `NullAway:ERROR` enabled locally and it was flagging some false positives, so I have not enabled it fully. IntelliJ also was warning for some of these, so I added the null check in this after the `isNonNullContext` callers. I may just switch back to a `void debugNullContext(\@Nullable InvocationContext)` and null check as that works and keeps the static analysis happy, code simple enough, and JIT should be happy.

Reply:
@cakofony I'm on the fence for just logging the `Class` object because it triggers my OCD as you get `class foo.bar.Baz` instead of `foo.bar.Baz`, so left the `safeClassName` in for now.

Went ahead and switched back to `debugNullContext` and cleaned up the few places that were preventing being able to fully enable `NullAway` at `ERROR`.

Reply:
Good point, I had not considered the “class “ prefix